browser_patches/firefox/juggler/content/FrameTree.js

```diff
@@ -452,72 +461,48 @@ class Frame {
-        return context;
-    }
-    return this.createIsolatedWorld(worldName);
+    for (const context of this._worldNameToContext.values())
+      this._runtime.destroyExecutionContext(context);
+    this._worldNameToContext.clear();
   }
 
   _addBinding(worldName, name, script) {
-    const executionContext = worldName ? this._getOrCreateIsolatedContext(worldName) : this._executionContext;
+    let executionContext = this._worldNameToContext.get(worldName);
+    if (worldName && !executionContext)
+      executionContext = this._createIsolatedContext(worldName);
     if (executionContext)
       executionContext.addBinding(name, script);
   }
 
-  _evaluateScriptSafely(executionContext, script) {
-    try {
-      let result = executionContext.evaluateScript(script);
-      if (result && result.objectId)
-        executionContext.disposeObject(result.objectId);
-    } catch (e) {
-      dump(`ERROR: ${e.message}\n${e.stack}\n`);
-    }
-  }
-
   _onGlobalObjectCleared() {
     const webSocketService = this._frameTree._webSocketEventService;
     if (this._webSocketListenerInnerWindowId)
       webSocketService.removeListener(this._webSocketListenerInnerWindowId, this._webSocketListener);
     this._webSocketListenerInnerWindowId = this.domWindow().windowGlobalChild.innerWindowId;
     webSocketService.addListener(this._webSocketListenerInnerWindowId, this._webSocketListener);
 
-    if (this._executionContext)
-      this._runtime.destroyExecutionContext(this._executionContext);
-    for (const world of this._isolatedWorlds.values())
-      this._runtime.destroyExecutionContext(world);
-    this._isolatedWorlds.clear();
+    for (const context of this._worldNameToContext.values())
+      this._runtime.destroyExecutionContext(context);
+    this._worldNameToContext.clear();
 
-    this._executionContext = this._runtime.createExecutionContext(this.domWindow(), this.domWindow(), {
+    this._worldNameToContext.set('', this._runtime.createExecutionContext(this.domWindow(), this.domWindow(), {
       frameId: this._frameId,
       name: '',
-    });
-    for (const {script, worldName} of this._frameTree._isolatedWorlds.values())
-      this.createIsolatedWorld(worldName);
-
-    // Add bindings before evaluating scripts.
-    for (const [id, {worldName, name, script}] of this._frameTree._bindings)
-      this._addBinding(worldName, name, script);
-
-    for (const script of this._frameTree._scriptsToEvaluateOnNewDocument.values())
-      this._evaluateScriptSafely(this._executionContext, script);
-    for (const {script, worldName} of this._frameTree._isolatedWorlds.values()) {
-      const context = worldName ? this._getOrCreateIsolatedContext(worldName) : this.executionContext();
-      this._evaluateScriptSafely(context, script);
+    }));
+    for (const [name, world] of this._frameTree._isolatedWorlds) {
+      if (name)
+        this._createIsolatedContext(name);
+      const executionContext = this._worldNameToContext.get(name || '');
```

Comment:
unless I'm missing something, the `name || ''` is not needed since `name` is guaranteed to be a string

Reply:
done